Fixed the problem that caused installation to fail.

decfrr opened the pull request:
## What
Fix #126 

## Why
The problem was due to an unset version limit for numpy.
With [the release of version 2.0.0](https://pypi.org/project/numpy/2.0.0/) on the 16th of this month, which contained some destructive changes, the build of this package was failing.
By setting the upper version limit, the build will succeed.

## Tasks
- [x] Passed all tests
- [x] Successful build

```
Successfully built nnmnkwii pysptk fastdtw
Installing collected packages: tqdm, threadpoolctl, scipy, joblib, fastdtw, decorator, cython, scikit-learn, pysptk, nnmnkwii
Successfully installed cython-3.0.10 decorator-5.1.1 fastdtw-0.3.4 joblib-1.4.2 nnmnkwii-0.1.2+67ff5ff pysptk-0.2.2 scikit-learn-1.5.0 scipy-1.13.1 threadpoolctl-3.5.0 tqdm-4.66.4
```

r9y9 commented:
Looks good to me. Can you apply the changes in #128 to fix CI failures?

```
Run actions/setup-python@v1
Error: Version 3.9 with arch x64 not found
```

decfrr commented:
Thanks for the confirmation.
I have merged and imported the contents of the patch. Please confirm.

r9y9 commented:
umm, looks like numpy 2.0.0 is installed on CI 🤔 

```
Using cached numpy-2.0.0-cp39-cp39-macosx_14_0_arm64.whl (5.2 MB)
```

decfrr commented:
I guess when we try to build this package using non version 2.0 numpy, we need to remove the cache in advance.

decfrr commented:
Also, rather than specifying less than a specific version, I thought I should specify a version less than 2.0.
I don't think this is a point to be particular about since it is a primary response, but I think it is better this way, so I changed it.

decfrr commented:
> I guess when we try to build this package using non version 2.0 numpy, we need to remove the cache in advance.

I too had a problem with ci downloading numpy 2.0. I will investigate.

r9y9 commented:
I guess I found the reason: `setup_requires` is deprecated

> https://setuptools.pypa.io/en/latest/userguide/dependency_management.html

> In previous versions of setuptools, this used to be accomplished with the setup_requires keyword but is now considered deprecated in favor of the [PEP 517](https://peps.python.org/pep-0517/) style described above. To peek into how this legacy keyword is used, consult our [guide on deprecated practice (WIP)](https://setuptools.pypa.io/en/latest/deprecated/index.html).

so, what about removing `setup_requires` and instead add the numpy requirement in `install_requires`?

decfrr commented:
By using `install_requires` instead of `setup_requires`, the problem of numpy version had been solved.

Also, some test problems had occured, I fixed them.

1. `np.int` type has been deprecated
https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
2. change error type RuntimeError -> ValueError of wrong number test
3. Clarified `librosa.effects.time_stretch` argument

And I found a error of deprecated field `np.object`, I fixed it.

decfrr commented:
I see that the test fails in 3.7.
I'd like you to run the test on other versions just to be sure.

r9y9 commented:
Testing other versions/platforms with github actions (https://github.com/r9y9/nnmnkwii/pull/128) and it is working OK too. I'll merge this PR after all CI gets green by either

1. dropping python 3.7 (easiest; I'm OK. It's already EOL https://devguide.python.org/versions/)
2. another idea if you prefer

decfrr commented:
I have merged the changes of including dropping python 3.7 from ci.
It seems that all CI will complete jobs successfully.

r9y9 reviewed (approved):
LGTM, thanks!

r9y9 commented on tests/data/questions-radio_dnn_416.hed:
Sorry, I have one last comment. Is this permission change intentional? If not, could you revert it?

r9y9 replied:
never mind, this is my bad in #128... I will fix this after merging this PR

decfrr commented:
Thank you for valuable opportunity.
I appereciate your ongoing maintenance.
